Applications/Utils/FileConverter/MeshToRaster.cpp

```diff
@@ -0,0 +1,205 @@
+{
+    MathLib::Vector3 const n = MeshLib::FaceRule::getSurfaceNormal(&elem).getNormalizedVector();
+    MeshLib::Node const orig = *elem.getNode(0);
+    MathLib::Point3d const v{{node[0] - orig[0], node[1] - orig[1], node[2] - orig[2]}};
+    double const dist = n[0] * v[0] + n[1] * v[1] + n[2] * v[2];
+    return node[2] - dist * n[2];
+}
+
+int main(int argc, char* argv[])
+{
+    ApplicationsLib::LogogSetup logog_setup;
+
+    TCLAP::CmdLine cmd(
+        "Mesh to raster converter.\n"
+        "Rasterises a 2D mesh, pixel values are set to the elevation of a "
+        "regular grid superimposed on the mesh. If no mesh element is located "
+        "beneath  a pixel it is set to NODATA.\n\n"
+        "OpenGeoSys-6 software, version " +
+            BaseLib::BuildInfo::git_describe +
+            ".\n"
+            "Copyright (c) 2012-2019, OpenGeoSys Community "
+            "(http://www.opengeosys.org)",
+        ' ', BaseLib::BuildInfo::git_describe);
+    TCLAP::ValueArg<std::string> input_arg(
+        "i", "input-file", "Mesh input file (*.vtu, *.msh)", true, "", "string");
+    cmd.add(input_arg);
+    TCLAP::ValueArg<std::string> output_arg(
+        "o", "output-file", "Raster output file (*.asc)", true, "", "string");
+    cmd.add(output_arg);
+    TCLAP::ValueArg<double> cell_arg(
+        "c", "cellsize", "edge length of raster cells in result", false, 1, "real");
+    cmd.add(cell_arg);
+
+    cmd.parse(argc, argv);
+
+    INFO("Rasterising mesh...");
+    if (!input_arg.isSet())
+    {
+        ERR("No input file given. Please specify a mesh file (*.vtu, *.msh)");
+        return -1;
+    }
+
+    if (!output_arg.isSet())
+    {
+        ERR("No output file given. Please specify a raster file (*.asc)");
+        return -1;
+    }
+
+    std::unique_ptr<MeshLib::Mesh> const mesh (
+        MeshLib::IO::readMeshFromFile(input_arg.getValue()));
+    if (mesh == nullptr)
+    {
+        ERR("Error reading mesh file.");
+        return 1;
+    }
+    if (mesh->getDimension() != 2)
+    {
+        ERR("The programme requires a mesh containing two-dimensional elements "
+            "(i.e. triangals or quadrilaterals.");
+        return 2;
+    }
+
+    double const cellsize = (cell_arg.isSet()) ? cell_arg.getValue()
+                                               : mesh->getMinEdgeLength();
+    INFO ("Cellsize set to %f", cellsize);
+
+    std::size_t const nNodes (mesh->getNumberOfNodes());
+    std::vector<MeshLib::Node*> const& nodes_vec(mesh->getNodes());
+    GeoLib::AABB bounding_box(nodes_vec.begin(), nodes_vec.end());
+    MathLib::Point3d const& min(bounding_box.getMinPoint());
+    MathLib::Point3d const& max(bounding_box.getMaxPoint());
+    std::size_t const n_cols = static_cast<std::size_t>(std::ceil((max[0] - min[0]) / cellsize));
+    std::size_t const n_rows = static_cast<std::size_t>(std::ceil((max[1] - min[1]) / cellsize));
+
+    // raster header
+    std::ofstream out(output_arg.getValue(), std::ios::out);
```

Comment:
ofstream is already having the default argument for the open mode set to `std::ios::out`, no need to mention it here.

Reply:
:heavy_check_mark: